sklearn/decomposition/dict_learning.py

```diff
@@ -110,7 +110,8 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         clf = Lasso(alpha=alpha, fit_intercept=False, precompute=gram,
                     max_iter=max_iter, warm_start=True)
         clf.coef_ = init
-        clf.fit(dictionary.T, X.T)
+        # Copying X to avoid buffer source-array read only error
+        clf.fit(dictionary.T, X.copy().T)
```

Comment:
hum. That's a workaround but not nice. Can you add a test to see what's wrong with Lasso and read only X?